buildSrc/src/main/resources/META-INF/schemas/guide-metadata.schema.json

```
@@ -0,0 +1,206 @@
+      "type": "integer"
+    },
+    "prerequisites": {
+      "description": "The prerequisites of the guide",
+      "type": "array",
+      "items": {
+        "type": "string"
+      }
+    },
+    "cloud": {
+      "description": "The cloud service provider of the guide",
+      "type": "string"
+    },
+    "skipGradleTests": {
+      "description": "Set it to true to skip running the tests for the Gradle applications for the guide",
+      "type": "boolean"
+    },
+    "skipMavenTests": {
+      "description": "Set it to true to skip running the tests for the Maven applications for the guide",
+      "type": "boolean"
+    },
+    "asciidoctor": {
+      "description": "The guide asciidoctor file",
+      "type": "string"
+    },
+    "languages": {
+      "description": "The guide supported languages",
+      "type": "array",
+      "items": {
+        "type": "string",
+        "enum": ["java","groovy","kotlin"]
+      }
+    },
+    "tags": {
+      "description": "List of tags added to the guide",
+      "type": "array",
+      "items": {
+        "type": "string"
+      }
+    },
+    "buildTools": {
+      "description": "By default the code in the guide is generated for Gradle and Maven. If a guide is specific only for a build tool, define it here",
+      "type": "array",
+      "items": {
+        "type": "string",
+        "enum": ["maven","gradle"]
+      }
+    },
+    "testFramework": {
+      "description": "The guide's test framework. By default Java and Kotlin applications are tested with JUnit5 and Groovy applications with Spock",
+      "type": "string",
+      "enum": ["spock","junit"]
+    },
+    "zipIncludes": {
+      "description": "List of additional files to include in the generated zip file for the guide",
+      "type": "array",
+      "items": {
+        "type": "string"
+      }
+    },
+    "categories": {
+      "description": "The guide's categories",
+      "type": "array",
+      "items": {
+        "type": "string"
+      }
+    },
+    "slug": {
+      "description": "The guide's slug",
+      "type": "string"
+    },
+    "publish": {
+      "description": "Defaults to true for regular guides; set to false for partial/base guides",
+      "type": "boolean"
+    },
+    "base": {
+      "description": "Defaults to null; if set, indicates directory name of the base guide to copy before copying the current one",
+      "type": "string"
+    },
+    "env": {
+      "description": "The guide's environment variables",
+      "type": "object",
+      "properties": {
+        "AWS_SECRET_ACCESS_KEY": {
+          "description": "Amazon Web Services Access Key",
+          "type": "string"
+        },
+        "AWS_REGION": {
+          "description": "Amazon Web Services region",
+          "type": "string"
+        },
+        "AWS_ACCESS_KEY_ID": {
+          "description": "Amazon Web Services Key Id",
+          "type": "string"
+        }
```

This is more an example than the real properties of the `env` property. The idea is you can supply a custom map a JSON object with the environment variables necessary for the guide. 

```suggestion
        "AWS_SECRET_ACCESS_KEY": {
          "description": "Amazon Web Services Access Key",
          "type": "string"
        },
        "AWS_REGION": {
          "description": "Amazon Web Services region",
          "type": "string"
        },
        "AWS_ACCESS_KEY_ID": {
          "description": "Amazon Web Services Key Id",
          "type": "string"
        }
```